oauthenticator/gitlab.py

```diff
@@ -179,6 +179,11 @@ async def authenticate(self, handler, data=None):
 
         no_config_specified = not (is_group_specified or is_project_id_specified)
 
+        if (is_group_specified and user_in_group) or (
+            is_project_id_specified and user_in_project
```

Comment:
Can you add `allowed_users` to this condition, because we don't want to start populating it if it's empty and unused?